### PR TITLE
W&B App: Rewrite _index.md for custom charts

### PR DESCRIPTION
**Pull request by ngrayluna**
## Description

Rewrote custom charts landing page (_index.md).  After testing it out myself, I realized the rest of the page could use some TLC. It's by no means perfect, but it'll hopefully bridge some knowledge gaps.

Inspired by this ticket https://wandb.atlassian.net/browse/DOCS-1001

Among other things it explains:

* query editor (what each field means, how to get specific values from each field, etc.) <- never documented before
* query data sources
* adds actual code examples that users can actually copy and paste

## Testing

For newly added examples I created, ran these code examples myself. 

## Resources: 

* Reference docs: https://docs.wandb.ai/ref/python/custom-charts/
* This colab: https://colab.research.google.com/drive/1uXLKDmsYg7QMRVFyjUAlg-eZH2MW8yWH?usp=sharing
* Custom charts walkthrough (surprisingly useful) https://docs.wandb.ai/guides/app/features/custom-charts/walkthrough/
* This somewhat outdated video: https://www.youtube.com/watch?v=3-N9OV6bkSM

📄 **[View preview links for changed pages](https://github.com/wandb/docs/pull/1685#issuecomment-3354273213)**

### CODE REVIEW

File: content/en/guides/models/app/features/custom-charts/_index.md [hunk 1]
```diff
@@ -10,30 +10,165 @@ cascade:
 - url: guides/app/features/custom-charts/:filename
 ---
 
-Create custom charts in your W&B project. Log arbitrary tables of data and visualize them exactly how you want. Control details of fonts, colors, and tooltips with the power of [Vega](https://vega.github.io/vega/).
+Custom charts are visualizations you create explicitly. Use custom charts to visualize more complex data relationships and have greater control over the appearance and behavior of the chart.
```
Comment (mdlinville) [inline content/en/guides/models/app/features/custom-charts/_index.md]: I bet you could do without "more". Wdyt?

File: content/en/guides/models/app/features/custom-charts/_index.md [hunk 2]
```diff
@@ -10,30 +10,165 @@ cascade:
...
 
-* Code: Try an example [Colab Colab notebook](https://tiny.cc/custom-charts).
-* Video: Watch a [walkthrough video](https://www.youtube.com/watch?v=3-N9OV6bkSM).
-* Example: Quick Keras and Sklearn [demo notebook](https://colab.research.google.com/drive/1g-gNGokPWM2Qbc8p1Gofud0_5AoZdoSD?usp=sharing)
+To create a custom chart, you can either:
 
-{{< img src="/images/app_ui/supported_charts.png" alt="Supported charts from vega.github.io/vega" max-width="90%" >}}
+* [Build a chart in the W&B App](#create-a-custom-chart-in-the-wb-app) by pulling in data from your runs using a [GraphQL](https://graphql.org) query.
```
Comment (mdlinville) [inline content/en/guides/models/app/features/custom-charts/_index.md:17]:
```suggestion
* [Build a chart in the W&B App](#create-a-custom-chart-in-the-wb-app) m data from your runs using a [GraphQL](https://graphql.org) query.
```

File: content/en/guides/models/app/features/custom-charts/_index.md [hunk 3]
```diff
@@ -10,30 +10,165 @@ cascade:
...
+with wandb.init() as run:
+  offset = random.random()
+  for run_step in range(20):
+    run.log({
+        "acc": math.log(1 + random.random() + run_step) + offset,
+        "val_acc": math.log(1 + random.random() + run_step) + offset * random.random(),
+    })
+```
+
+The code snippet logs two metrics `acc` and `val_acc` at each training step (each `wandb.Run.log()` call). In the W&B App, navigate to the Workspace tab. Within the `Charts` panel, two line plots appear: one for `acc` and one for `val_acc`. The x-axis is the `step` value (automatically tracked by W&B), and the y-axis is the metric value you provide with `wandb.Run.log()`.
```
Comment (mdlinville) [inline content/en/guides/models/app/features/custom-charts/_index.md]:
```suggestion
This code logs two metrics `acc` and `val_acc` at each training step (each `wandb.Run.log()` call). In the W&B App, navigate to the Workspace tab. Within the `Charts` panel, two line plots appear: one for `acc` and one for `val_acc`. The x-axis is the `step` value (automatically tracked by W&B), and the y-axis is the metric value you provide with `wandb.Run.log()`.
```
More importantly, I think a stepped procedure is buried on this paragraph.

File: content/en/guides/models/app/features/custom-charts/_index.md [hunk 4]
```diff
@@ -10,30 +10,165 @@ cascade:
...
+Each call to `wandb.Run.log()` creates a new entry in the run's history. The `step` value is automatically tracked by W&B and increments with each log call.
+
+Once the code snippet completes, navigate to your project's workspace:
+
+<!-- 
+In the W&B App, navigate to the Workspace tab. Within the `Charts` panel, two line plots appear: one for `acc` and one for `val_acc`. The x-axis is the `step` value (automatically tracked by W&B), and the y-axis is the metric value you provide with `wandb.Run.log()`. -->
+
+<!-- Suppose you want to create a line plot of the accuracy values you logged (`acc`) vs step. To do this: -->
+
+1. Click on the **Add panels** button in the top right corner, then select **Custom chart**.
```
Comment (mdlinville) [inline content/en/guides/models/app/features/custom-charts/_index.md]:
```suggestion
1. Click the **Add panels** button in the top right corner, then select **Custom chart**.
```

File: content/en/guides/models/app/features/custom-charts/_index.md [hunk 5]
```diff
@@ -10,30 +10,165 @@ cascade:
...
+4. Within the chart editor, select `_step` for the **X** field and `acc` for the **Y** field. Optionally, set a title for the chart. Your settings should look similar to the following:
+{{< img src="/images/app_ui/custom-charts-query-example.png" alt="Custom line plot settings" max-width="90%" >}}
 
-1. **Log data**: From your script, log [config]({{< relref "/guides/models/track/config.md" >}}) and summary data.
-2. **Customize the chart**: Pull in logged data with a [GraphQL](https://graphql.org) query. Visualize the results of your query with [Vega](https://vega.github.io/vega/), a powerful visualization grammar.
-3. **Log the chart**: Call your own preset from your script with `wandb.plot_table()`.
 
-{{< img src="/images/app_ui/pr_roc.png" alt="PR and ROC curves" >}}
+Note that only the keys you specify in the query editor are available in the chart editor (below the query editor). If you do not see the expected keys, go back to the query editor and ensure you have selected the correct data source and specified the correct keys.
```
Comment (mdlinville) [inline content/en/guides/models/app/features/custom-charts/_index.md:77]: Maybe this would be better as an actual note since it might clear up confusion.

File: content/en/guides/models/app/features/custom-charts/_index.md [hunk 6]
```diff
@@ -10,30 +10,165 @@ cascade:
...
+* `summary`: A single-value metrics that summarize a run. It's populated by the final value of metrics logged with `wandb.Run.log()` or by directly updating the run.summary object. Think of it as a key-value store for your run's final results.
+* `history`: Time series data logged with `wandb.Run.log()`. Each call to `log()` creates a new row in the history table. This is ideal for tracking metrics that change over time, like training and validation loss.
+* `summaryTable`: A table of summary metrics. It's populated by logging a `wandb.Table` to the `summary` field. This is useful for logging metrics that are best represented in a tabular format, like confusion matrices or classification reports.
+* `historyTable`: A table of time series data. It's populated by logging a `wandb.Table` to the `history` field. This is useful for logging complex metrics that change over time, like per-epoch evaluation metrics.
+
+To summarize, `summary` and `history` are the general locations for your run data, while `summaryTable` and `historyTable` are the specific query types needed to access tabular data stored in those respective locations.
```
Comment (mdlinville) [inline content/en/guides/models/app/features/custom-charts/_index.md]: I don't love summarize and summary together

File: content/en/guides/models/app/features/custom-charts/_index.md [hunk 7]
```diff
@@ -10,30 +10,165 @@ cascade:
...
+* `runSets`: Top-level object, representing the set of runs you are currently viewing or have filtered in the UI.
+* `summary(...)`, `history(...)`, `summaryTable(...)`, `historyTable(...)`: This tells the query to fetch data from the respective objects of each run.
+* `keys: [""]`: An array of strings where each string is the name (or key) of the metric or object you want to retrieve.
 
 ## Log charts from a script
 
-### Builtin presets
+You can programmatically create a custom chart from your script by logging a `wandb.Table` of the data you want to visualize, then calling `wandb.plot.*` to create the chart. 
+
+For example consider the following code snippet:
```
Comment (mdlinville) [inline content/en/guides/models/app/features/custom-charts/_index.md]:
```suggestion
For example, consider the following code snippet:
```

File: content/en/guides/models/app/features/custom-charts/_index.md [hunk 8]
```diff
@@ -10,30 +10,165 @@ cascade:
...
+```
+
+Within the W&B app, navigate to the **Workspace** tab. Within the `Custom Charts` panel, there are three charts with the following titles: **Scatter Plot**, **Histogram**, and **Line Plot**. These correspond to the three charts created in the script above. The x-axis and y-axis are set to the columns specified in the `wandb.plot.*` function calls (`height` and `step`). 
 
-W&B has a number of builtin chart presets that you can log directly from your script. These include line plots, scatter plots, bar charts, histograms, PR curves, and ROC curves.
+The following image shows the three custom charts created from the script:
+
+{{< img src="/images/app_ui/custom-charts-script-plots.png" alt="Custom charts from script" max-width="90%" >}}
+
+### Builtin chart types
```
Comment (mdlinville) [inline content/en/guides/models/app/features/custom-charts/_index.md]: Do we tend to use the unhyphenated form? I associate it with bash.

**Review by mdlinville — CHANGES_REQUESTED** [review]
Great updates. Not directly approving in case automerge is on so you can evaluate. None are blockers really, so feel free to dismiss the review and merge after you decide whether or not to make adjustments.

File: content/en/guides/models/app/features/custom-charts/_index.md (hunk 2 above)
Comment (mdlinville) [reply content/en/guides/models/app/features/custom-charts/_index.md:17]: Oops that was supposed to be "from" not just "m".
